File: src/Frontend/src/models/Destination.js
export class Destination {
  constructor(data = {}) {
    this.id = data.id || '';
    this.name = data.name || '';
    this.location = data.location || '';
    this.arrivalDate = data.arrivalDate ? new Date(data.arrivalDate) : null;
    this.departureDate = data.departureDate ? new Date(data.departureDate) : null;
    this.notes = data.notes || '';
    this.travelPlanId = data.travelPlanId || '';
  }
}

File: src/Services/TravelService/Models/Destination.cs
using System;

namespace TravelService.Models
{
    public class Destination
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Location { get; set; } = string.Empty;
        public DateTime ArrivalDate { get; set; }
        public DateTime DepartureDate { get; set; }
        public string Notes { get; set; } = string.Empty;
        public Guid TravelPlanId { get; set; }
    }
}
